crates/swc_ecma_transforms_base/src/feature.rs

```diff
@@ -0,0 +1,253 @@
+
+    /// `transform-computed-properties`
+    ComputedProperties,
+
+    /// `transform-for-of`
+    ForOf,
+
+    /// `transform-sticky-regex`
+    StickyRegex,
+
+    /// `transform-dotall-regex`
+    DotAllRegex,
+
+    /// `transform-unicode-regex`
+    UnicodeRegex,
+
+    /// `transform-spread`
+    Spread,
+
+    /// `transform-parameters`
+    Parameters,
+
+    /// `transform-destructuring`
+    Destructuring,
+
+    /// `transform-block-scoping`
+    BlockScoping,
+
+    /// `transform-typeof-symbol`
+    TypeOfSymbol,
+
+    /// `transform-new-target`
+    NewTarget,
+
+    /// `transform-regenerator`
+    Regenerator,
+
+    /// `transform-exponentiation-operator`
+    ExponentiationOperator,
+
+    /// `transform-async-to-generator`
+    AsyncToGenerator,
+
+    /// `proposal-async-generator-functions`
+    AsyncGeneratorFunctions,
+
+    /// `proposal-object-rest-spread`
+    ObjectRestSpread,
+
+    /// `proposal-unicode-property-regex`
+    UnicodePropertyRegex,
+
+    /// `proposal-json-strings`
+    JsonStrings,
+
+    /// `proposal-optional-catch-binding`
+    OptionalCatchBinding,
+
+    /// `transform-named-capturing-groups-regex`
+    NamedCapturingGroupsRegex,
+
+    /// `transform-member-expression-literals`
+    MemberExpressionLiterals,
+
+    /// `transform-property-literals`
+    PropertyLiterals,
+
+    /// `transform-reserved-words`
+    ReservedWords,
+
+    /// `proposal-export-namespace-from`
+    ExportNamespaceFrom,
+
+    /// `proposal-nullish-coalescing-operator`
+    NullishCoalescing,
+
+    /// `proposal-logical-assignment-operators`
+    LogicalAssignmentOperators,
+
+    /// `proposal-optional-chaining`
+    OptionalChaining,
+
+    /// `proposal-class-properties`
+    ClassProperties,
+
+    /// `proposal-numeric-separator`
+    NumericSeparator,
+
+    /// `proposal-private-methods`
+    PrivateMethods,
+
+    /// `proposal-class-static-block`
+    ClassStaticBlock,
+
+    /// `proposal-private-property-in-object`
+    PrivatePropertyInObject,
+
+    /// `transform-unicode-escapes`
+    UnicodeEscapes,
+
+    /// `bugfix/transform-async-arrows-in-class`
+    BugfixAsyncArrowsInClass,
+
+    /// `bugfix/transform-edge-default-parameters`
+    BugfixEdgeDefaultParam,
+
+    /// `bugfix/transform-tagged-template-caching`
+    BugfixTaggedTemplateCaching,
+
+    /// `bugfix/transform-safari-id-destructuring-collision-in-function-expression`
+    BugfixSafariIdDestructuringCollisionInFunctionExpression,
+
+    /// `bugfix/transform-edge-function-name`
+    BugfixTransformEdgeFunctionName, // TODO
+
+    /// `bugfix/transform-safari-block-shadowing`
+    BugfixTransformSafariBlockShadowing, // TODO
+
+    /// `bugfix/transform-safari-for-shadowing`
+    BugfixTransformSafariForShadowing, // TODO
+
+    /// `bugfix/transform-v8-spread-parameters-in-optional-chaining`
+    BugfixTransformV8SpreadParametersInOptionalChaining, // TODO
+}
+
+pub type FeatureSet = Rc<RefCell<AHashSet<Feature>>>;
```

Comment:
I'm a bit worried about `Rc<RefCell<T>>`.

Reply:
This type is not used by compat transforms, right?

Reply:
I'm asking this because some passes transform file in parallel.

Reply:
Is `Rc<RefCell<T>>` really required?

Reply:
I need to modify it in build function (preset or ES version target) and use it in module pass.
See 
https://github.com/swc-project/swc/blob/058e09f4d65d13cf4563c3f77f9285088e4ee6eb/crates/swc_ecma_preset_env/src/lib.rs#L71-L77

And
https://github.com/swc-project/swc/blob/fb299ce89f94e5ba2abf82b1e11b1c54fb42e92b/crates/swc/src/builder.rs#L198-L203

Is it possible to just use `&mut `?

Reply:
I think so? It can accept &mut AHashSet instead.

Reply:
I will try the bitflag which is easy to copy and modify.

Reply:
Done.